samples/gibbs_ensemble/lj_gibbs_client.py

```diff
@@ -0,0 +1,105 @@
+#
+# Copyright (C) 2013-2019 The ESPResSo project
+#
+# This file is part of ESPResSo.
+#
+# ESPResSo is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# ESPResSo is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+"""
+Client part of the Gibbs ensemble simulation. This script handles the
+simulation boxes and communicates the energies to the host. The Monte-Carlo
+part of the simulation is done by the :file:`gibbs_ensemble_socket.py` script.
+"""
+
+import argparse
+
+import numpy as np
+
+import espressomd
+import gibbs
+
+espressomd.assert_features("LENNARD_JONES")
+
+seed = None
+
+LJ_EPSILON = 1.0
+LJ_SIGMA = 1.0
+LJ_CUTOFF = 2.5
+LJ_SHIFT = 0
+
+
+parser = argparse.ArgumentParser()
+parser.add_argument('-s', '--seed', type=int)
+parser.add_argument('-H', '--host', type=str, default='localhost')
+parser.add_argument('-p', '--port', type=int, required=True)
```

Comment:
for a single required argument it should be avoided to mark it as optional. If you provide proper help messages with the ```help``` keyword argument to ```add_argument``` the meaning of the positional argument is not ambiguous.